modello-maven-plugin/src/main/java/org/codehaus/modello/maven/ModelloVelocityMojo.java

```diff
@@ -0,0 +1,117 @@
+package org.codehaus.modello.maven;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import java.io.File;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Properties;
+import java.util.stream.Collectors;
+
+import org.apache.maven.plugins.annotations.LifecyclePhase;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.codehaus.modello.plugin.velocity.VelocityGenerator;
+
+/**
+ * Creates files from the model using Velocity templates.
+ * <p>
+ * This mojo can be given a list of templates and a list of parameters.
+ * Each template from the {@link #templates} property will be run with the following context:
+ * <ul>
+ *    <li>{@code version}: the version of the model to generate</li>
+ *    <li>{@code model}: the modello model</li>
+ *    <li>{@code Helper}: a {@link org.codehaus.modello.plugin.velocity.Helper} object instance</li>
+ *    <li>any additional parameters specified using the {@link #params} property</li>
+ * </ul>
+ * The output file is controlled from within the template using the {@code #MODELLO-VELOCITY#REDIRECT}
```

Comment:
I find the term "redirect" confusing: makes me think HTTP redirect.
Can we rename to something like "OUTPUT"?